Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/view/ProgressLoadingButton.kt

```diff
@@ -54,27 +53,46 @@ class ProgressLoadingButton @JvmOverloads constructor(
                 applyTo(parentLayout)
             }
 
-            val loadingText = getText(R.styleable.ProgressLoadingButton_loadingText) ?: ""
             val loading = getBoolean(R.styleable.ProgressLoadingButton_isLoading, false)
-            defaultText = getText(R.styleable.ProgressLoadingButton_buttonText).toString()
+            defaultText = getText(R.styleable.ProgressLoadingButton_buttonText)?.toString() ?: ""
             binding.apply {
                 defaultButton.text = defaultText
-                loadingButton.text = loadingText
                 isLoading = loading
             }
         }
     }
 
     var isLoading: Boolean = false
         set(value) {
+            if (isActive) {
+                binding.apply {
+                    isClickable = !value
+                    defaultButton.isClickable = !value
+                    isEnabled = !value
+                    defaultButton.isEnabled = !value
+                    defaultButton.text = if (value) "" else defaultText
+                    progressIndicator.isVisible = value
+                }
+                field = value
+            }
+        }
+
+    var isActive: Boolean = true
+        set(value) {
+            if (!value) {
```

Comment:
How come you do not set the field first? Just wandering as you set isLoading here which checks the value of isActive, which has not been updated at this point